File: src/main/java/xyz/chunshengyuan/mall/controller/OpenController.java
package xyz.chunshengyuan.mall.controller;

import org.springframework.web.bind.annotation.*;
import xyz.chunshengyuan.mall.configuration.RequiredRole;
import xyz.chunshengyuan.mall.constrant.UserRole;
import xyz.chunshengyuan.mall.exceptions.AdminLoginFailedException;
import xyz.chunshengyuan.mall.exceptions.WxRedirectException;
import xyz.chunshengyuan.mall.model.BaseResponse;
import xyz.chunshengyuan.mall.model.bo.DetailUser;
import xyz.chunshengyuan.mall.model.vo.AdminLoginForm;
import xyz.chunshengyuan.mall.model.vo.WxUserForm;
import xyz.chunshengyuan.mall.service.UserService;
import xyz.chunshengyuan.mall.utils.RequestContext;

import javax.annotation.Resource;

/**
 * @author leemaster
 * @Title: OpenController
 * @Package xyz.chunshengyuan.mall.controller
 * @Description:
 * @date 2019-07-1002:17
 */
@RestController
@RequestMapping("/open")
public class OpenController {

    @Resource
    UserService userService;

    /**
     * 管理端登陆
     *
     * @param form
     * @return
     * @throws AdminLoginFailedException
     */
    @PostMapping("/admin/login")
    public BaseResponse adminLogin(@RequestBody AdminLoginForm form) throws AdminLoginFailedException {
        return BaseResponse.succes(
                userService.adminLogin(
                        form.getPhone(),
                        form.getPassword()
                )
        );
    }

    /**
     * 用户登陆
     *
     * @return
     */
    @PostMapping("/wx/login")
    public BaseResponse wxLogin(@RequestParam("code") String code) throws WxRedirectException, AdminLoginFailedException {
        return BaseResponse.succes(userService.wxAccountLogin(code));
    }

    /**
     * 用户信息补全
     *
     * @return
     */
    @PutMapping("/wx/complete")
    public BaseResponse wxComplete(@RequestBody WxUserForm form) {
        return BaseResponse.succes(userService.wxAccountRegister(form));
    }
}

File: src/main/kotlin/xyz/chunshengyuan/mall/controller/OpenController.kt
package xyz.chunshengyuan.mall.controller

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import xyz.chunshengyuan.mall.mapper.UserMapper
import xyz.chunshengyuan.mall.model.BaseResponse
import xyz.chunshengyuan.mall.model.UserInfo
import xyz.chunshengyuan.mall.model.success
import xyz.chunshengyuan.mall.service.CommonUserService

/**
 * @Title: OpenController
 * @Package xyz.chunshengyuan.mall.controller
 * @Description:
 * @author leemaster
 * @date 2019-07-06 23:07
 */

@RestController
@RequestMapping("/open/wx")
class WxController @Autowired constructor(

){


}


@RestController
@RequestMapping("/open/admin")
class AdminController @Autowired constructor(
    private val commonUserService: CommonUserService
){


    data class AdminLoginForm(
        val phone:String? = null,
        val password:String? = null
    )
    @PostMapping("/login")
    fun adminLogin(@RequestBody form: AdminLoginForm): BaseResponse<UserInfo> = success(commonUserService.adminUserLogin(form.phone ?: "",form.password ?: ""))



}